src/iotile/models/properties.ts

```diff
@@ -0,0 +1,8 @@
+export class Properties {
+
+  constructor(datas) {
+    for (let data of datas) {
+      this[data.name] = data.value;
```

Comment:
I am making the changes so that `Properties` is an array of `Property` instead.

In the example above, does the `foo` and `bar` represent the `name` and `value` field respectively on the `Property` model?